////////////////////////////////////////////////////////////////////////////////
// Copyright (c) 2017 Dimitry Ishenko
// Contact: dimitry (dot) ishenko (at) (gee) mail (dot) com
//
// Distributed under the GNU GPL license. See the LICENSE.md file for details.

////////////////////////////////////////////////////////////////////////////////
#ifndef GUI_WINDOW_HPP
#define GUI_WINDOW_HPP

////////////////////////////////////////////////////////////////////////////////
#include <QWidget>

////////////////////////////////////////////////////////////////////////////////
namespace gui
{

namespace Ui { class window; }

////////////////////////////////////////////////////////////////////////////////
class window : public QWidget
{
    Q_OBJECT

public:
    explicit window(QWidget* parent = nullptr);
    ~window();

signals:
    void closing();

protected:
    virtual void closeEvent(QCloseEvent*) override;

private:
    Ui::window* ui;

    bool open_ = false;
    bool connected_ = false;

    void serial_open();
    void serial_close();
    void serial_settings();

    void casparcg_connect();
    void casparcg_disconnect();
    void casparcg_scan();
};


////////////////////////////////////////////////////////////////////////////////
}

////////////////////////////////////////////////////////////////////////////////
#endif
